data/gui/widget/text_box_default.cfg

```diff
@@ -36,6 +36,16 @@
 	[/rectangle]
 #enddef
 
+#
+# The preprocessor can't process GUI__FONT_COLOR_DISABLED__DEFAULT as part of string
```

Comment:
@Vultraz Isn't it more likely the fact that the macro definition itself contains quotes?

The macro is defined like this:
```
#define GUI__FONT_COLOR_DISABLED__DEFAULT
#arg ALPHA
255 #endarg
"128, 128, 128, {ALPHA}" #enddef
```

Both of the `#end`s are at the end of a line, so unless there's a bug somewhere, there should be no newlines in that expansion.

You want to use it like this:

```"(
	if(text = '' and hint_text != '', hint_color, reg_color) where
		hint_color = [{GUI__FONT_COLOR_DISABLED__DEFAULT}],
		reg_color = [{COLOR}]
)"```

Isolating that one line, we have:

```"...[{GUI__FONT_COLOR_DISABLED__DEFAULT}]..."```

So that would expand to:

```"...["128, 128, 128, 255"]..."```

I'm not sure how the parser handles the case of a closing quote being immediately followed by something other than a comment or whitespace, or an opening quote being immediately preceded by something other than whitespace. There may be an actual bug with how that's handled.

However… a key point you can notice in that expansion is that the colour is _unquoted_. The colour is in quotes to begin with, so it can't just be dropped wholesale into a quote – it needs to be _concatenated_.

Thus, I believe the following should work:

```"...[" + {GUI__FONT_COLOR_DISABLED__DEFAULT} + "]..."```

Or, to expand out the entire formula again:

```"(
	if(text = '' and hint_text != '', hint_color, reg_color) where
		hint_color = [" + {GUI__FONT_COLOR_DISABLED__DEFAULT} + "],
		reg_color = [{COLOR}]
)"```